### PR TITLE
Added a exclusive column list , which can be used for salesforce configuration with huge list of columns. 

### PR DESCRIPTION
**Pull request by dvenkateshappa**
Added a exclusive column list , which can be used for salesforce configuration with huge list of columns.

### CODE REVIEW

**Comment by dvenkateshappa** [comment]
@narasimhareddyv 
@zliu41 

Could you please review the changes.

File: gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java [hunk 1]
```diff
@@ -312,6 +312,7 @@
   public static final String SOURCE_QUERYBASED_HOUR_COLUMN = "source.querybased.hour.column";
   public static final String SOURCE_QUERYBASED_SKIP_HIGH_WATERMARK_CALC = "source.querybased.skip.high.watermark.calc";
   public static final String SOURCE_QUERYBASED_QUERY = "source.querybased.query";
+  public static final String SOURCE_QUERYBASED_EXCLUDED_COLUMNS = "source.querybased.excludedcolumns";
```
Comment (zliu41) [inline gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java]: `excludedcolumns` → `excluded.columns`.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 2]
```diff
@@ -115,6 +115,15 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
       columnListInQuery = Utils.getColumnListFromQuery(inputQuery);
     }
 
+	// Deepak added the below code for building the exclusion list
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: Please remove this comment, the git history will record who added what.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 3]
```diff
@@ -148,8 +157,12 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
           if (inputQuery == null || columnListInQuery == null || (columnListInQuery.size() == 1 && columnListInQuery
               .get(0).equals("*")) || (columnListInQuery.size() >= 1 && this
               .isMetadataColumn(columnName, columnListInQuery))) {
-            this.columnList.add(columnName);
-            columnArray.add(jsonObject);
+            // Deepak added the below condition for checking in exclusion list
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: ditto

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 4]
```diff
@@ -115,6 +115,15 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
       columnListInQuery = Utils.getColumnListFromQuery(inputQuery);
     }
 
+	// Deepak added the below code for building the exclusion list
+
+	String excludedColumns = workUnit.getProp(ConfigurationKeys.SOURCE_QUERYBASED_EXCLUDED_COLUMNS);
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: Please format correctly. This line and the next line have different indentations.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 5]
```diff
@@ -115,6 +115,15 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
...
+    List<String> columnListExcluded = new ArrayList<String>();
+    
+    if (Strings.isNullOrEmpty(inputQuery)  && !Strings.isNullOrEmpty(excludedColumns)) {
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: Can you add some explanation here, why do you need to check `Strings.isNullOrEmpty(inputQuery)`

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 6]
```diff
@@ -115,6 +115,15 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
...
+    	columnListExcluded =  Arrays.asList(excludedColumns.toLowerCase().replaceAll(" ", "").split(","));
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: Please use `com.google.common.base.Splitter.splitToList`

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 7]
```diff
@@ -115,6 +115,15 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
       columnListInQuery = Utils.getColumnListFromQuery(inputQuery);
     }
 
+	// Deepak added the below code for building the exclusion list
+
+	String excludedColumns = workUnit.getProp(ConfigurationKeys.SOURCE_QUERYBASED_EXCLUDED_COLUMNS);
+    List<String> columnListExcluded = new ArrayList<String>();
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: Please use `ImmutableList.<String> of()`

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java [hunk 8]
```diff
@@ -148,8 +155,11 @@ public void extractMetadata(String schema, String entity, WorkUnit workUnit)
           if (inputQuery == null || columnListInQuery == null || (columnListInQuery.size() == 1 && columnListInQuery
               .get(0).equals("*")) || (columnListInQuery.size() >= 1 && this
               .isMetadataColumn(columnName, columnListInQuery))) {
-            this.columnList.add(columnName);
-            columnArray.add(jsonObject);
+             if (!columnListExcluded.contains(columnName.trim().toLowerCase()))
+             {
+        		   this.columnList.add(columnName);
+        		   columnArray.add(jsonObject);
+      		  }
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/restapi/RestApiExtractor.java]: This line has a different indentation, and it seems it contains tabs. Could you make sure your code conforms to LinkedIn's coding conventions (https://iwww.corp.linkedin.com/wiki/cf/display/ENGS/Java+Coding+Conventions) by downloading the XML and use it in Eclipse.

**Comment by dvenkateshappa** [comment]
Hi Ziyang, 

Could you please approve the changes, so that we can use the merged code soon.

Thanks
Deepak

**Comment by zliu41** [comment]
@dvenkateshappa looks good, can you merge these 4 commits into one?

**Comment by dvenkateshappa** [comment]
Thanks Ziyang.

**Pull request merged**